include/ddsrouter/types/ParticipantType.hpp

```diff
@@ -0,0 +1,74 @@
+
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file ParticipantType.hpp
+ */
+
+#ifndef _DDSROUTER_TYPES_PARTICIPANTTYPE_HPP_
+#define _DDSROUTER_TYPES_PARTICIPANTTYPE_HPP_
+
+#include <map>
+#include <string>
+
+namespace eprosima {
+namespace ddsrouter {
+
+//! Participant types available
+enum ParticipantType
+{
+    INVALID,    //! Invalid Participant Type
+    VOID,       //! Void Participant Type
+};
+
+/**
+ * @brief Object that handles the creation of ParticipantTypes from a string
+ */
+class ParticipantTypeFactory
+{
+public:
+
+    /**
+     * @brief Create a Participant Type regarding the string argument in lower case
```

Comment:
```suggestion
     * @brief Creates a Participant Type regarding the string argument in lower case
```

Reply:
I think it is better impersonal notation